ipa-core/src/protocol/ipa_prf/boolean_ops/share_conversion_aby.rs

```diff
@@ -437,16 +468,19 @@ mod tests {
 
             let [res0, res1, res2] = world
                 .semi_honest(records.into_iter(), |ctx, records| async move {
+                    let c_ctx = ctx.set_total_records((COUNT + CONV_CHUNK - 1) / CONV_CHUNK);
+                    let validator = &c_ctx.dzkp_validator((COUNT + CONV_CHUNK - 1) / CONV_CHUNK);
+                    let m_ctx = validator.context();
                     seq_join(
-                        ctx.active_work(),
+                        m_ctx.active_work(),
                         process_slice_by_chunks(&records, |idx, chunk| {
-                            let ctx = ctx.clone();
                             let match_keys =
                                 BitDecomposed::transposed_from(&*chunk).unwrap_infallible();
-                            convert_to_fp25519::<_, CONV_CHUNK, PRF_CHUNK>(
-                                ctx.set_total_records((COUNT + CONV_CHUNK - 1) / CONV_CHUNK),
+                            convert_to_fp25519::<_, _, CONV_CHUNK, PRF_CHUNK>(
+                                validator.clone(),
+                                idx,
                                 RecordId::from(idx),
-                                match_keys,
+                                vec![match_keys],
```

Comment:
It's probably worth changing this test to do two chunks, to exercise the proof batching in a context other than end-to-end tests.